### ISSUE TITLE
conflicting report of RV_ABSOLUTE_VALUE_OF_HASHCODE on two semantically similar code

### ISSUE THREAD

**Issue by csrma** [spotbugs/spotbugs#3447]
When using the following code, SpotBugs correctly flags `RV_ABSOLUTE_VALUE_OF_HASHCODE`.
```java
class HashCodeBugExample {

    public int showBug(String input) {
        int hashcode = input.hashCode();
        int absHashcode = Math.abs(hashcode); // BUG: RV: Bad attempt to compute absolute value of signed 32-bit hashcode (RV_ABSOLUTE_VALUE_OF_HASHCODE)
        
        // Unreachable switch statement
        int unreachableValue = -1;
        switch (unreachableValue) {
            case 1:
                System.out.println("This case is unreachable.");
                break;
            default:
                break;
        }

        return absHashcode;
    }

    public static void main(String[] args) {
        HashCodeBugExample example = new HashCodeBugExample();

        System.out.println("Absolute hashcode: " + example.showBug("polygenelubricants"));
    }
}
```

However, when `-1` is replaced by `Integer.MIN_VALUE`, SpotBugs no longer detects the rule violation, although it is semantically the same code.
```java
class HashCodeBugExample {

    public int showBug(String input) {
        int hashcode = input.hashCode();
        int absHashcode = Math.abs(hashcode); // BUG: RV: Bad attempt to compute absolute value of signed 32-bit hashcode (RV_ABSOLUTE_VALUE_OF_HASHCODE)
        
        // Unreachable switch statement
        int unreachableValue = Integer.MIN_VALUE;;
        switch (unreachableValue) {
            case 1:
                System.out.println("This case is unreachable.");
                break;
            default:
                break;
        }

        return absHashcode;
    }

    public static void main(String[] args) {
        HashCodeBugExample example = new HashCodeBugExample();

        System.out.println("Absolute hashcode: " + example.showBug("polygenelubricants"));
    }
}
```
Note that it only happens when using `Integer.MIN_VALUE` and nothing else. Using `Integer.MAX_VALUE` or any other variable also leads to correct detection by SpotBugs.

SpotBugs version: `4.9.2`
Java: `Temurin openjdk 21.0.7`
Ran from CLI as the following two commands:
```bash
export SPOTBUGS_JAR=/path/to/spotbugs.jar
```
```bash
javac HashCodeBugExample.java && \
java -jar $SPOTBUGS_JAR \
    -low \
    -effort:max \
    HashCodeBugExample.class
```

Expectation: Should detect the issue in both cases. How the `unreachableValue` variable is assigned should not affect the hashcode calculation part.

**Comment by gtoison** [comment]
The detector is a bit crude and does not report the bug when it sees that the MIN_VALUE constant was loaded. This is most probably to avoid false positives when the code correctly handles the case when the hash had that value.

I think it would be complicated to detect those situations where the negative value was handled correctly.
